include/boost/url/experimental/detail/router.hpp

```diff
@@ -39,8 +39,17 @@ struct match_any_t {};
 // A dynamic path segment
 class segment_template
 {
+    enum class modifier : std::uint8_t
```

Comment:
why trot out a clumsy declared type when you could use a built-in type like unsigned char? Please explain the rationale.

Reply:
why use a library type when you could use a built-in?